diff --git a/boards/native/native_sim/native_sim.dts b/boards/native/native_sim/native_sim.dts
index 459399fbf64..a497b5dbb03 100644
--- a/boards/native/native_sim/native_sim.dts
+++ b/boards/native/native_sim/native_sim.dts
@@ -37,6 +37,7 @@
 		i2s-node0 = &i2s_rxtx;
 		spi-0 = &spi0;
 		led0 = &led0;
+		buzzer0 = &buzzer0;
 		rtc = &rtc;
 	};
 
@@ -49,6 +50,11 @@
 		};
 	};
 
+	buzzer0: buzzer {
+		compatible = "native-sim-buzzer";
+		status = "disabled";
+	};
+
 	cpus {
 		#address-cells = <1>;
 		#size-cells = <0>;
diff --git a/drivers/buzzer/CMakeLists.txt b/drivers/buzzer/CMakeLists.txt
index 07f40a56a6f..4f170e2037a 100644
--- a/drivers/buzzer/CMakeLists.txt
+++ b/drivers/buzzer/CMakeLists.txt
@@ -5,3 +5,8 @@ zephyr_library()
 
 zephyr_library_sources_ifdef(CONFIG_BUZZER_GPIO buzzer_gpio.c)
 zephyr_library_sources_ifdef(CONFIG_BUZZER_PWM  buzzer_pwm.c)
+
+if(CONFIG_BUZZER_NATIVE_SIM)
+  zephyr_library_sources(buzzer_native_sim.c)
+  target_sources(native_simulator INTERFACE buzzer_native_sim_bottom.c)
+endif()
diff --git a/drivers/buzzer/Kconfig b/drivers/buzzer/Kconfig
index d6f158b38b7..3daee161625 100644
--- a/drivers/buzzer/Kconfig
+++ b/drivers/buzzer/Kconfig
@@ -21,5 +21,6 @@ config BUZZER_INIT_PRIORITY
 
 rsource "Kconfig.gpio"
 rsource "Kconfig.pwm"
+rsource "Kconfig.native_sim"
 
 endif # BUZZER
